src/osd/OSD.cc

```diff
@@ -6582,7 +6582,8 @@ void OSD::handle_scrub(MOSDScrub *m)
 
 bool OSD::scrub_random_backoff()
 {
-  bool coin_flip = (rand() % 3) == whoami % 3;
+  bool coin_flip = (rand() / (double)RAND_MAX >=
+		    cct->_conf->osd_scrub_backoff_ratio);
```

Comment:
The purpose of using whoami before must have been that if all the OSDs booted at the same time and the values from rand() produced the same sequence on all OSDs (seeding off the clock?), then all OSDs would backoff at the same time.  I'm confused about rand() and thread safety.  I found that srand() is being called multiple times on OSD start but not sure what that means.  I filed http://tracker.ceph.com/issues/18873